### PR TITLE
114 past runs backend

### LINKED ISSUE #114: Make the past runs fragment retrieve and show the actual list of runs of the user

### PR DESCRIPTION
**Pull request by CristianSafta**
Changed the backend of the historyFragment to load asynchronously the list of runs from the database or cache,

Closes #114

### CODE REVIEW

File: app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/HistoryFragment.kt
Comment (HoLuDyEr) [inline app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/HistoryFragment.kt:1]: It is better to put logic inside
`override fun onViewCreated(view: View, savedInstanceState: Bundle?) {...}`
as`onCreateView` should only be used for inflating.

**Review by HoLuDyEr — CHANGES_REQUESTED** [review]
Overall this PR is good, it only needs small changes.

File: app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/HistoryFragment.kt [hunk 1]
```diff
@@ -26,38 +27,21 @@ class HistoryFragment : Fragment(R.layout.fragment_history) {
...
-    }
+        val userModelCached: UserModelCached by activityViewModels()
+
+        val runHistoryObserver = object : Observer<List<Run>> {
+            override fun onChanged(value: List<Run>) {
+                if (value != null) {
+                    runsAdapter.updateRunsData(value)
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (CristianSafta) [reply app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/HistoryFragment.kt]: Done, thanks for the suggestion

**Review by HoLuDyEr — APPROVED** [review]
Thank you for the changes, it is ready to be merged

**Pull request merged**